synapse/storage/databases/main/room.py

```diff
@@ -1361,14 +1427,27 @@ async def upsert_room_on_join(self, room_id: str, room_version: RoomVersion):
         # mark the room as having an auth chain cover index.
         has_auth_chain_index = await self.has_auth_chain_index(room_id)
 
+        create_event = None
+        for e in auth_events:
```

Comment:
I originally had this using `self.get_create_event_for_room(room_id)` but changed it to passing in the `auth_events` because the current state ID's weren't persisted yet when this is called in [`do_invite_join`](https://github.com/matrix-org/synapse/blob/ecd823d766fecdd7e1c7163073c097a0084122e2/synapse/handlers/federation.py#L1681-L1693).

We could [call `_persist_auth_tree` before we call `upsert_room_on_join`](https://github.com/matrix-org/synapse/blob/ecd823d766fecdd7e1c7163073c097a0084122e2/synapse/handlers/federation.py#L1690-L1697) to make them available AFAICT 🤷  but that order may be more fragile than I am assuming.